p2p/router.go

```diff
@@ -792,11 +760,10 @@ func (r *Router) routePeer(peerID NodeID, conn Connection, sendQueue queue) {
 
 	e1 := <-errCh
 	err := e1
-	_ = conn.Close()
-	sendQueue.close()
-
 	e2 := <-errCh
```

Comment:
Are you sure this won't block? If there's an error on the receivePeer then this won't stop the sendPeer routine from continuing to run right?

Reply:
the channel is buffered, so I think it's fine.

Reply:
Hmm maybe I'm not sure about what the desired behavior is. When one of these go routines fails we want the other to also stop right?

Reply:
This might be an artifact of the behavior of the transport/connection, but I've never observed one half of the connection dying without the other end dying. I suspect that there's a stop channel in the connection implementation that is responsible for this. 

Fundamentally, this isn't a change in the PR, I'm just trying to surface both errors in log messages for better clarity.